src/Sylius/Bundle/AdminBundle/Action/Account/ResetPasswordAction.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\AdminBundle\Action\Account;
+
+use Sylius\Bundle\AdminBundle\Form\Model\PasswordReset;
+use Sylius\Bundle\AdminBundle\Form\Type\ResetPasswordType;
+use Sylius\Bundle\CoreBundle\Message\Admin\Account\ResetPassword;
+use Symfony\Component\Form\FormFactoryInterface;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\Session\Flash\FlashBagInterface;
+use Symfony\Component\Messenger\MessageBusInterface;
+use Symfony\Component\Routing\RouterInterface;
+use Twig\Environment;
+
+final class ResetPasswordAction
+{
+    public function __construct(
+        private FormFactoryInterface $formFactory,
+        private MessageBusInterface $messageBus,
+        private FlashBagInterface $flashBag,
+        private RouterInterface $router,
+        private Environment $twig,
+    ) {
+    }
+
+    public function __invoke(Request $request, string $token): Response
+    {
+        $form = $this->formFactory->create(ResetPasswordType::class);
+
+        $form->handleRequest($request);
+        if ($form->isSubmitted() && $form->isValid()) {
+            /** @var PasswordReset $formData */
+            $formData = $form->getData();
+            $resetPasswordCommand = new ResetPassword(
+                $token,
+                $formData->getPassword(),
+            );
+
+            $this->messageBus->dispatch($resetPasswordCommand);
```

Comment:
I know it's not a lot of logic, but I would still put it in a separate service, for an important reason - better extendability. With the proposed implementation it would be hard to do anything _around_ command dispatching or enrich the command with some additional data... or change the command to the other one :dancer: